### ISSUE TITLE
Blank Notification

### ISSUE THREAD

**Issue by raymond-dillon** [nhsx/COVID-19-app-Android-BETA#1]
**Describe the bug**
Blank notification after installing the app

**To Reproduce**
Steps to reproduce the behaviour:
1. Follow the app on boarding process
2. Notification appears
3. No text in notification

**Expected behaviour**
Text appears in the notification

![715](https://user-images.githubusercontent.com/28100585/81333445-8b9f6b00-909c-11ea-9c49-64af18f508f7.png)

**Desktop (please complete the following information):**
 N/A

**Smartphone (please complete the following information):**
 - Device: OnePlus 5
 - OS: Android
 - Version 10

**Additional context**
N/A

**Comment by benabrahamson** [comment]
I think this notification is intentional - likely to keep the app alive in the backgroud.

You might be specifically referring to the lack of text though, in which case perhaps it might be nice to have a message there such as "The app is working as expected", or maybe "Follow regular advice" (indicating that there is not a self-isolation alert for you).

**Comment by hithomasmorelli** [comment]
Just out of interest, is this an "ongoing notification" that can't be cleared? Or is it one that can just be cleared?

**Comment by benabrahamson** [comment]
> Just out of interest, is this an "ongoing notification" that can't be cleared? Or is it one that can just be cleared?

Can't be cleared

**Comment by jcampbell05** [comment]
Smart way of keeping android app alive

**Comment by rosiebye** [comment]
Will people be ok with a notification permanently sitting on their phone? I personally don't like them and have uninstalled apps that do this. I might be willing to put up with it due to the nature of this app but it could discourage people from using the app.

An alternative is to use a background service for the bluetooth scanning and ask the user to disable battery optimizations. See https://developer.android.com/training/monitoring-device-state/doze-standby#support_for_other_use_cases.

**Comment by hithomasmorelli** [comment]
Just my opinion, but I would prefer to have an ongoing notification that told me the app was working normally - I might worry otherwise!

Also re. asking user to disable battery optimizations, for every extra step the user has to take to use an app, the more users that will give up due to added hassle - in my opinion (again) it's probably best to keep that to a minimum

**Comment by Wild1145** [comment]
I believe it's intentional, I've seen it with a few other apps that like to be kept alive in the background as well, only way to get rid of the notification is to force kill the app, which is obviously counter-productive. It might be worth however making it clear to users that the notification is signaling that the app is running and not just a orphaned notification.

**Comment by jcampbell05** [comment]
It’s the recommended google android method for doing this and it’s better to be explicit

**Comment by LDuncAndroid** [comment]
Definitely intentional.

This is the relevant Android doc: https://developer.android.com/guide/components/services#Foreground

And here's where it's being used in the app: https://github.com/nhsx/COVID-19-app-Android-BETA/blob/43a167f8dba422fd9001b64f9c4fd82275abb1c8/app/src/main/java/uk/nhs/nhsx/sonar/android/app/ble/BluetoothService.kt#L68

**Comment by edent** [comment]
Thanks. We'll look into putting in some more explanatory text in the notification.

**Comment by edent** [comment]
I'm pasting this message in every active GitHub issue, so you may receive duplicate notifications.

Today, I'm happy to announce that NHSX has released the full git commit history for the Isle of Wight Beta apps.

* Android - https://github.com/nhsx/COVID-19-app-Android-BETA
* iOS - https://github.com/nhsx/COVID-19-app-iOS-BETA

As discussed, we have redacted API keys, sensitive domain names, and some of the developers' personal details. I am still waiting on final approval to publish the server-side code.

I would like to personally thank the community for your comments, bug reports, and vulnerability disclosures. They all went into helping the development process.

The beta trial of this app has now ended and we've moved to the next phase of app development. It is our intention to publish the source code of future apps as the binaries are released to the public.

Once again, thank you for being part of this.

Terence Eden
Head of Open Technology - NHSX
